### PR TITLE
redis self-query

### PR DESCRIPTION
**Pull request by baskaryan**

### CODE REVIEW

**Comment by baskaryan** [comment]
cc @Spartee @tylerhutcherson

File: libs/langchain/langchain/retrievers/self_query/redis.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    RedisFilterExpression,
+    RedisFilterField,
+    RedisFilterOperator,
+    RedisNum,
+    RedisTag,
+    RedisText,
+)
+from langchain.vectorstores.redis.schema import RedisModel
+
+_COMPARATOR_TO_BUILTIN_METHOD = {
```
Comment (Spartee) [inline libs/langchain/langchain/retrievers/self_query/redis.py:24]: put in ``constants.py``?

File: libs/langchain/langchain/retrievers/self_query/redis.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                )
+            )
+        else:
+            right = operation.arguments[1].accept(self)
+        redis_operator = (
+            RedisFilterOperator.OR
+            if operation.operator == Operator.OR
+            else RedisFilterOperator.AND
+        )
+        return RedisFilterExpression(operator=redis_operator, left=left, right=right)
```
Comment (Spartee) [inline libs/langchain/langchain/retrievers/self_query/redis.py:89]: while not designed to be constructed like this, this should work. I can dig more into the logic when I'm not on a plane, but this is the part i'd be sure to dig into.

**Review by Spartee — COMMENTED** [review]
Minor comments, can't fully do this review on a plane, will have to be when I land.

A couple comments in the meantime
- Should we have an option to silence the warning about metadata schema mismatch?
- The jupyter notebook seems to have an issue finding the metadata for the ``genre`` field. (hard to view on github)


Will have a more complete review later

File: libs/langchain/langchain/retrievers/self_query/redis.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+from langchain.vectorstores.redis.schema import RedisModel
+
+_COMPARATOR_TO_BUILTIN_METHOD = {
+    Comparator.EQ: "__eq__",
+    Comparator.NE: "__ne__",
+    Comparator.LT: "__lt__",
+    Comparator.GT: "__gt__",
+    Comparator.LTE: "__le__",
+    Comparator.GTE: "__ge__",
+    Comparator.CONTAIN: "__eq__",
```
Comment (nfcampos) [inline libs/langchain/langchain/retrievers/self_query/redis.py:31]: Hmmm is this one right?

Comment (baskaryan) [reply libs/langchain/langchain/retrievers/self_query/redis.py:31]: i belieeeeve for tags == is what's used to check if a tag is in a list, but correct me if im wrong @Spartee

Comment (Spartee) [reply libs/langchain/langchain/retrievers/self_query/redis.py:31]: Yes, ``==`` is the comparator.

We did this because ``!=`` also works for ``not in`` but also because ``__contains__`` (i.e. "in") works and ``not in`` doesn't. This is because we're overriding the ``__eq__`` behavior``. 

Originally we thought about just using the fluent style, but we've seen so many users mess that up in Redis-Py. It's also not very discoverable. 

Lots of other reasons too.

TLDR: Looks right.

**Review by Spartee — APPROVED** [review]
Approving given a couple asks

1. Clarify why its throwing

```
UserWarning: The predict_and_parse method is deprecated, instead pass an output parser directly to LLMChain.
```

2. Please, right an integration test. We're going to start running these and testing coverage. Would love it if you could test all the metadata types. lmk if you want help with this.

Also, what if we just added a ``description`` field (optional) to the schema? could we just automatically generate the ``AttributeInfo``?
